src/inc/til/ticket_lock.h

```diff
@@ -53,4 +53,89 @@ namespace til
         std::atomic<uint32_t> _next_ticket{ 0 };
         std::atomic<uint32_t> _now_serving{ 0 };
     };
+
+    struct recursive_ticket_lock
+    {
+        struct recursive_ticket_lock_suspension
+        {
+            constexpr recursive_ticket_lock_suspension(recursive_ticket_lock& lock) noexcept :
+                _lock{ lock }
+            {
+            }
+
+            ~recursive_ticket_lock_suspension()
+            {
+                if (_owner)
+                {
+                    _lock._lock.lock(); // lock-lock-lock lol
+                    _lock._owner.store(_owner, std::memory_order_relaxed);
+                    _lock._recursion = _recursion;
```

Comment:
same question about storing `_recursion` actually. A suspended lock should not resume with the same recursion depth, because mightn't someone have acquired it in the meantime?

Reply:
I'm imagining a contrived scenario:

```
{
 auto susp = l.suspend();
 l.lock();
}
```

Or, is the point of suspending it so that your entire call stack releases it and gets it back later? is there a way to enforce that?

Reply:
Good point... This is definitely a bit risky...

> Or, is the point of suspending it so that your entire call stack releases it and gets it back later?

Yeah that's the idea. I added it for situations like `ControlCore::_terminalPlayMidiNote` in this PR. Without recursive locking this would be easy, but with it we need to somehow recover the recursion count. Hence the need for this `suspension` struct.

> is there a way to enforce that?

We can make suspension a tiny bit more expensive, which seems fine, since it's not widely used.